### ISSUE TITLE
A JOIN SQL with aggregate function(COUNT) in stockLevelTransaction()

### ISSUE THREAD

**Issue by Lordess** [Lordess/benchmarksql_for_mysql#2]
# phenonmenon

  orderStatusTransaction() has the SQL belows. MyCat (1.6.5 BETA) can not parse it well, because it is a JOIN with aggregate function COUNT(). Also we cannot just wait before MyCat fix it or do somethings.

``` Java
.prepareStatement("SELECT COUNT(DISTINCT (s_i_id)) AS stock_count"
      + " FROM order_line, stock"
      + " WHERE ol_w_id = ?"
      + " AND ol_d_id = ?"
      + " AND ol_o_id < ?"
      + " AND ol_o_id >= ? - 20"
      + " AND s_w_id = ?"
      + " AND s_i_id = ol_i_id"
      + " AND s_quantity < ?");
```

#  workaround

  It seems that the current MyCat is not good at subquery, so it is suggested to split the query into 2-step simple CRUD query like belows:

1. Get the list of **ol_i_id** from *order_line*

``` Java
.prepareStatement("SELECT ol_i_id"
      + " FROM order_line"
      + " WHERE ol_w_id = ?"
      + " AND ol_d_id = ?"
      + " AND ol_o_id < ?"
      + " AND ol_o_id >= ? - 20");		

// concate ol_i_id with comma, like this: "ol_i_id1, ol_i_id2, ..."

```

2. Do SELECT on *stock* with **ol_i_id list** from above

``` Java

.prepareStatement("SELECT COUNT(DISTINCT (s_i_id)) AS stock_count"
    + " FROM stock"
    + " WHERE s_w_id = ?"
    + " AND s_quantity < ?"
    + " AND s_i_id IN (" + in_list + ")");		
```

**Comment by Lordess** [comment]
fixed by commit e251140305e789a74325f3fcae6a79cf675c7670

**Comment by Lordess** [comment]
fixed by commit e251140305e789a74325f3fcae6a79cf675c7670
